### PR TITLE
Get Stories Available for Review by User

### PR DESCRIPTION
**Pull request by andrewyguo**
## Notion ticket link

[List all stories available to be reviewed by User](https://www.notion.so/uwblueprintexecs/List-all-stories-available-to-be-reviewed-by-User-40702ca6c0ef4daab38a9e2d997aed03)

## Implementation description
*  Created an additional story query that retrieves all of the stories available to be reviewed
*  Given a level and a language, query will return all story translations in that language that can be reviewed by someone with these credentials 

## Steps to test
1. Seed database with different stories and seed database with different story_translations that have a reviewer_id of None 
Example: 
`INSERT INTO stories (title, description, youtube_link, level, translated_languages) VALUES ('Kevin Eats Caviar for Lunch', 'He complains its too cheap', 'link', 4, '{"GERMAN", "ENGLISH_UK"}');`
`INSERT INTO stories (title, description, youtube_link, level, translated_languages) VALUES ('Kevin Eats Gold', 'He complains its too cheap', 'link', 2, '{"GERMAN", "MANDARIN"}');`
Insert into StoryTranslations: 
`INSERT INTO story_translations (story_id, language, stage, translator_id) VALUES (1,  'FRENCH', 'REVIEW', 1);`
`INSERT INTO story_translations (story_id, language, stage, translator_id) VALUES (2,  'ARABIC', 'REVIEW', 1);`
`INSERT INTO story_translations (story_id, language, stage, translator_id) VALUES (3,  'ARABIC', 'REVIEW', 1);`
`INSERT INTO story_translations (story_id, language, stage, translator_id) VALUES (2,  'GERMAN', 'REVIEW', 1);`
`INSERT INTO story_translations (story_id, language, stage, translator_id) VALUES (1,  'ENGLISH_US', 'START', 1);`
`INSERT INTO story_translations (story_id, language, stage, translator_id) VALUES (1,  'RUSSIAN', 'PUBLISH', 1);`

2. Call the query with various levels and languages and see returned list of story_translations

## What should reviewers focus on?
* Ensuring that the query returns story_translations as expected

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters:  `docker exec -it <backend-container-id> /bin/bash -c "black . && isort --profile black ."`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/python/app/services/interfaces/story_service.py [hunk 1]
```diff
@@ -50,3 +50,14 @@ def get_story_translations(self, user_id, translate):
         :rtype: list of StoryTranslationResponseDTO's
         """
         pass
+
+    @abstractmethod
+    def get_story_available_for_review(self, level, language):
+        """
+        Return a list of stories available to be reviewed by user
+        :param level: level of user 
+        language: language being searched for
```
Comment (gaoxk) [inline backend/python/app/services/interfaces/story_service.py:59]: nit: `language: language being searched for` -> `param language: language being searched for`

File: backend/python/app/services/interfaces/story_service.py [hunk 2]
```diff
@@ -50,3 +50,14 @@ def get_story_translations(self, user_id, translate):
         :rtype: list of StoryTranslationResponseDTO's
         """
         pass
+
+    @abstractmethod
+    def get_story_available_for_review(self, level, language):
```
Comment (gaoxk) [inline backend/python/app/services/interfaces/story_service.py:55]: nit: can you order params like the `get_stories_available_for_translation` fn? i.e. language and then level

File: backend/python/app/services/implementations/story_service.py [hunk 3]
```diff
@@ -120,3 +120,14 @@ def get_story_translations(self, user_id, translator):
...
             self.logger.error(str(error))
             raise error
+
+    def get_story_available_for_review(self, level, language):
+
+        stories = (
+            Story.query.join(StoryTranslation)
+            .filter(Story.level <= level)
+            .filter(StoryTranslation.language == language)
+            .filter(StoryTranslation.stage == "REVIEW")
```
Comment (gaoxk) [inline backend/python/app/services/implementations/story_service.py:130]: wrong filter: you're looking for something like `StoryTranslation.review_id == none`

BTW: a story translation that is available for review is in `TRANSLATION` stage, not `REVIEW` stage. The story is in `REVIEW` stage when a reviewer is assigned.

File: backend/python/app/services/implementations/story_service.py [hunk 4]
```diff
@@ -120,3 +120,14 @@ def get_story_translations(self, user_id, translator):
         except Exception as error:
             self.logger.error(str(error))
             raise error
+
+    def get_story_available_for_review(self, level, language):
```
Comment (gaoxk) [inline backend/python/app/services/implementations/story_service.py:124]: ditto nit about param order

File: backend/python/app/graphql/schema.py [hunk 5]
```diff
@@ -80,5 +86,8 @@ def resolve_stories_available_for_translation(root, info, language, level):
     def resolve_story_translations_by_user(root, info, user_id, translator):
         return resolve_story_translations_by_user(root, info, user_id, translator)
 
+    def resolve_story_available_for_review_by_user(root, info, level, language):
```
Comment (gaoxk) [inline backend/python/app/graphql/schema.py:89]: ditto nit about param order

File: backend/python/app/graphql/schema.py [hunk 6]
```diff
@@ -47,6 +48,11 @@ class Query(graphene.ObjectType):
     users = graphene.Field(graphene.List(UserDTO))
     user_by_id = graphene.Field(UserDTO, id=graphene.Int())
     user_by_email = graphene.Field(UserDTO, email=graphene.String())
+    story_available_for_review_by_user = graphene.Field(
+        graphene.List(StoryTranslationResponseDTO),
+        level=graphene.Int(),
```
Comment (gaoxk) [inline backend/python/app/graphql/schema.py:53]: ditto nit about param order

File: .gitignore [hunk 7]
```diff
@@ -8,3 +8,5 @@
 **/.DS_Store
 **/nodemailer.config.ts
 .vscode
+package-lock.json
+package.json
```
Comment (gaoxk) [inline .gitignore:12]: Is this part of the changes needed for eslint on vscode?

**Review by gaoxk — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 7 above)
Comment (andrewyguo) [reply .gitignore:12]: When I configured my eslint, it created these 2 files that had some of my local settings so I decided to gitignore it.

File: backend/python/app/graphql/schema.py (hunk 5 above)
Comment (gaoxk) [inline backend/python/app/graphql/schema.py:89]: Parameter order matters: right now the imported `resolve_story_available_for_review_by_user` is being passed a string as level and an integer as language. the function in `story_query`.py is expecting a language and then levl

Did you test out the mutation? Open up localhost:5000/graphql and try
```
{
  storyAvailableForReviewByUser(language: "HINDI", level:2){
    storyId,
    language
  }
}
```

File: backend/python/app/graphql/types/story_type.py [hunk 8]
```diff
@@ -49,6 +49,7 @@ class CreateStoryTranslationRequestDTO(graphene.InputObjectType):
     language = graphene.String(required=True)
     stage = graphene.Field(StageEnum, default_value="TRANSLATE")
 
+
```
Comment (gaoxk) [inline backend/python/app/graphql/types/story_type.py:52]: is this from the linter?

File: backend/python/app/services/interfaces/story_service.py [hunk 9]
```diff
@@ -50,3 +50,14 @@ def get_story_translations(self, user_id, translate):
         :rtype: list of StoryTranslationResponseDTO's
         """
         pass
+
+    @abstractmethod
+    def get_story_available_for_review(self, language, level):
```
Comment (gaoxk) [inline backend/python/app/services/interfaces/story_service.py:55]: nit: rename to `get_story_translations_available_for_review`

File: backend/python/app/graphql/queries/story_query.py [hunk 10]
```diff
@@ -15,3 +15,7 @@ def resolve_stories_available_for_translation(root, info, language, level):
 
 def resolve_story_translations_by_user(root, info, user_id, translator):
     return services["story"].get_story_translations(user_id, translator)
+
+
+def resolve_story_available_for_review_by_user(root, info, language, level):
```
Comment (gaoxk) [inline backend/python/app/graphql/queries/story_query.py:20]: nit: rename to `resolve_story_translations_available_for_review`
dont forget to change the name in all uses of the function

**Review by gaoxk — CHANGES_REQUESTED** [review]

File: backend/python/app/graphql/types/story_type.py (hunk 8 above)
Comment (andrewyguo) [reply backend/python/app/graphql/types/story_type.py:52]: Think so D:

File: backend/python/app/graphql/schema.py (hunk 5 above)
Comment (andrewyguo) [reply backend/python/app/graphql/schema.py:89]: Ah so I think what happened was I forgot to commit the changes I made to schema.py. It worked on my local system but it won't work with the current changes.

File: backend/python/app/services/implementations/story_service.py [hunk 11]
```diff
@@ -120,3 +120,30 @@ def get_story_translations(self, user_id, translator):
         except Exception as error:
             self.logger.error(str(error))
             raise error
+
+    def get_story_translations_available_for_review(self, language, level):
+        try:
+            stories = (
```
Comment (gaoxk) [inline backend/python/app/services/implementations/story_service.py:126]: nit: `stories =` -> `return`
(copy pattern from  `get_story_translations`)

**Review by gaoxk — APPROVED** [review]
🚢

**Pull request merged**